pkgs/development/compilers/gcc/patches/default.nix

```diff
@@ -284,6 +298,11 @@ in
   })
 ]
 
+# The clang-based assembler used in darwin.cctools-llvm (LLVM >11) does not support piping input.
+# Fortunately, it does not exhibit the problem GCC has with the cctools assembler.
+# This patch can be dropped should darwin.cctools-llvm ever implement support.
+++ optional (!atLeast7 && hostPlatform.isDarwin && lib.versionAtLeast (lib.getVersion stdenv.cc) "12") ./4.9/darwin-clang-as.patch
+
```

Comment:
The changes to `gcc/patches/default.nix` and the patches themselves look okay; if you apply the two minor changes I suggested above and put them in a separate PR targeted to `master` I will approve+merge it.  Then we can focus on the stuff in `default.nix`.

Reply:
Which ones am I splitting off? If I can drop adb07a447b76325a54ee289fe181af91a64e6e04, then the only commit that doesn’t patches is f1f4d827cdbaae8bf7f5d1e8e8271c4adefe91b4. Wouldn’t it make more sense to split that one off and retarget this PR to master?